Add cookie auth instructions to C8Run

conceptualshark opened the pull request:
## Description

To use any component API with C8Run, cookie authentication is required. The C8Run package already includes the environment variables necessary for cookie auth to work. v1 and v2 endpoints are noted separately to make sure the demo URLs are appropriate, and to note that some v2 endpoints are not yet fully enabled.

This also removes other mentions of cookie auth where C8Run is used for the preference of this setup, and updates the download link in preparation to the new patch.

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

<!-- Camunda maintains 18 months of versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

<!-- All changes require either an Engineering review or technical writer review. **Many require both!** -->

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). -->

akeller reviewed (dismissed):
This looks good! We also did extensive testing and validation via https://camunda.slack.com/archives/C07RB6W8URX/p1728658936123909

conceptualshark commented:
@akeller This should be ready for re-review:
- Updated the download links to the C8Run patch with the presumed URL 
- Removed the previous notes about authentication to rely on this single source of information for both v1 and v2 apis

The `curl` examples used are from the other versions of the guide, but I've updated them to use a non-alpha v2 endpoint. I haven't hit them yet myself (they differ slightly in structure from the examples used in testing, but the endpoints are the same). I'm going to do that now, but would also appreciate additional validation - or I can just update them to use identical syntax as the testing.

akeller commented on docs/self-managed/setup/deploy/local/c8run.md:
````diff
@@ -98,6 +96,55 @@ To add a custom Connector:
+1. Log in as user 'demo' and store the cookie in the file `cookie.txt`:
+
+```shell
+curl -c cookie.txt -X POST 'http://localhost:8080/api/login?username=demo&password=demo'
````
I see inconsistent flags (`-c`, `-X`, `-b`), which are optional AFAIK, but the calls are successful. I would prefer we remove flags for clarity unless we need them. WDYT @pepopowitz?

pepopowitz replied:
+1 to consistency across the calls. 

There are a couple other things that I think help readability on example curl requests, which I tried to implement on the auth rewrites ([example](https://docs.camunda.io/docs/apis-tools/tasklist-api-rest/tasklist-api-rest-authentication/?environment=saas)): 

1. One argument per line, using `\` to tell the shell that more is coming
2. Long-form flags (e.g. `--cookie-jar` instead of `-c`) because seeing the full name can help people who don't know what the short-form translates to (read: me 😅)

So in this example, something like: 

```sh
curl --request POST 'http://localhost:8080/api/login?username=demo&password=demo' \
  --cookie-jar cookie.txt 
```

In the next example, something like: 
```sh
curl --request POST 'http://localhost:8080/v1/process-definitions/search'  \
  --cookie cookie.txt
  --header 'Content-Type: application/json' \
  --data-raw '{}`
```

conceptualshark replied:
Thank you! I've gone ahead and updated this section to use the clearer formatting and flags.

akeller commented on docs/guides/react-components/_install-c8run.md:
```diff
@@ -13,7 +11,7 @@ If no version of Java is found, follow your chosen installation's instructions f
 ### Install and start Camunda 8 Run
 
-1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.0) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
+1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.1) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
```
```suggestion
1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.2) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
```

akeller commented on docs/self-managed/setup/deploy/local/c8run.md:
```diff
@@ -38,7 +36,7 @@ If no version of Java is found, follow your chosen installation's instructions f
 ## Install and start Camunda 8 Run
 
-1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.0) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
+1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.1) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
```
```suggestion
1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.2) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
```

akeller commented on versioned_docs/version-8.6/guides/react-components/_install-c8run.md:
```diff
@@ -13,7 +11,7 @@ If no version of Java is found, follow your chosen installation's instructions f
 ### Install and start Camunda 8 Run
 
-1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.0) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
+1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.1) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
```
```suggestion
1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.2) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
```

akeller commented on versioned_docs/version-8.6/self-managed/setup/deploy/local/c8run.md:
```diff
@@ -38,7 +36,7 @@ If no version of Java is found, follow your chosen installation's instructions f
 ## Install and start Camunda 8 Run
 
-1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.0) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
+1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.1) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
```
```suggestion
1. Download the [latest release of Camunda 8 Run](https://github.com/camunda/camunda/releases/tag/c8run-8.6.2) for your operating system and architecture. Opening the .tgz file extracts the Camunda 8 Run script into a new directory.
```

conceptualshark commented:
With 8.6.2 released, I've removed the hold label from this PR.
